tests/x11/keyboard_layout_gnome.pm

```diff
@@ -0,0 +1,66 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+
+# Summary: Test if gnome login manager honors keyboard layout changes
+# Maintainer: Dominik Heidler <dheidler@suse.de>
+
+use base "x11test";
+use strict;
+use testapi;
+use utils;
+use power_action_utils 'power_action';
+
+sub reboot {
+    my ($self, %args) = @_;
+    $args{setnologin} //= 0;
+    power_action('reboot', textmode => 1);
+    $self->wait_boot(nologin => $args{setnologin}, forcenologin => $args{setnologin});
+}
+
+sub run {
+    my $self             = shift;
+    my $kbdlayout_script = "changekbd.sh";
+
+    # login
+    select_console('root-console');
+
+    # disable autologin
+    assert_script_run "sed -i.bak '/^DISPLAYMANAGER_AUTOLOGIN=/s/=.*/=\"\"/' /etc/sysconfig/displaymanager";
+
+    # set german keyboard layout (y and z are switched on this layout)
+    type_string "echo \"test \\\$1 == qwerty && loadkeys us-intl && echo done > /dev/$serialdev\" > $kbdlayout_script\n";
+    assert_script_run "cat $kbdlayout_script";
```

Comment:
If you want to show the content of the script for crosschecking then I recommend to pipe through "tee" in the line above instead

Reply:
I would still need some command after the `type_string` to trigger a screenshot at the right time.